csharp/ql/src/experimental/dataflow/flowsources/AuthCookie.qll

```diff
@@ -0,0 +1,192 @@
+    result = src
+  )
+}
+
+/**
+ * Similar to getAValueForProp, checks if the given property was explicitly set to a value.
+ */
+predicate isPropertySet(ObjectCreation oc, string prop) {
+  exists(Assignment a, PropertyAccess pa |
+    a.getLValue() = pa and
+    pa.getTarget().hasName(prop) and
+    DataFlow::localExprFlow(oc, pa.getQualifier())
+  )
+  or
+  exists(Assignment a, PropertyAccess pa, Expr src |
+    a.getLValue() = pa and
+    pa.getTarget().hasName(prop) and
+    a.getRValue() = src and
+    DataFlow::localExprFlow(src,
+      oc.getInitializer().(ObjectInitializer).getAMemberInitializer().getRValue())
+  )
+}
+
+/**
+ * Tracks if a callback used in `OnAppendCookie` sets `Secure` to `true`.
+ */
+class OnAppendCookieSecureTrackingConfig extends OnAppendCookieTrackingConfig {
+  OnAppendCookieSecureTrackingConfig() { this = "OnAppendCookieSecureTrackingConfig" }
+
+  override string propertyName() { result = "Secure" }
+}
+
+/**
+ * Tracks if a callback used in `OnAppendCookie` sets `HttpOnly` to `true`.
+ */
+class OnAppendCookieHttpOnlyTrackingConfig extends OnAppendCookieTrackingConfig {
+  OnAppendCookieHttpOnlyTrackingConfig() { this = "OnAppendCookieHttpOnlyTrackingConfig" }
+
+  override string propertyName() { result = "HttpOnly" }
+}
+
+/**
+ * Tracks if a callback used in `OnAppendCookie` sets a cookie property to `true`.
+ */
+abstract class OnAppendCookieTrackingConfig extends DataFlow::Configuration {
+  bindingset[this]
+  OnAppendCookieTrackingConfig() { any() }
+
+  /**
+   * Specifies the cookie property name to track.
+   */
+  abstract string propertyName();
+
+  override predicate isSource(DataFlow::Node source) {
+    exists(PropertyWrite pw, Assignment delegateAssign, Callable c |
+      pw.getProperty().getName() = "OnAppendCookie" and
+      pw.getProperty().getDeclaringType() instanceof MicrosoftAspNetCoreBuilderCookiePolicyOptions and
+      delegateAssign.getLValue() = pw and
+      (
+        exists(LambdaExpr lambda |
+          delegateAssign.getRValue() = lambda and
+          lambda = c
+        )
+        or
+        exists(DelegateCreation delegate |
+          delegateAssign.getRValue() = delegate and
+          delegate.getArgument().(CallableAccess).getTarget() = c
+        )
+      ) and
+      c.getParameter(0) = source.asParameter()
+    )
+  }
+
+  override predicate isSink(DataFlow::Node sink) {
+    exists(PropertyWrite pw, Assignment a |
+      pw.getProperty().getDeclaringType() instanceof MicrosoftAspNetCoreHttpCookieOptions and
+      pw.getProperty().getName() = propertyName() and
+      a.getLValue() = pw and
+      exists(Expr val |
+        DataFlow::localExprFlow(val, a.getRValue()) and
+        val.getValue() = "true"
+      ) and
+      sink.asExpr() = pw.getQualifier()
+    )
+  }
+
+  override predicate isAdditionalFlowStep(DataFlow::Node node1, DataFlow::Node node2) {
+    node2.asExpr() = any(PropertyRead other | other.getQualifier() = node1.asExpr())
```

Comment:
Just checking: This adds flow from _any_ object to a property read on that object; is that the intention?